File: temp/fix_mixed_quotes.py
import os
import re

slides_dir = r"D:/新课开发/企业大学/对内/1.企业大学重生：从内训中心到知识资产与智能体孵化枢纽的角色转型/PPT演示文稿/slides"

for f in ['slide-94.js', 'slide-117.js', 'slide-140.js', 'slide-139.js', 'slide-141.js']:
    path = os.path.join(slides_dir, f)
    if not os.path.exists(path):
        continue
    with open(path, 'r', encoding='utf-8') as file:
        content = file.read()

    original = content

    # Fix: slide.addText('「", { -> slide.addText("「", {
    content = content.replace("slide.addText('「\", {", 'slide.addText("「", {')

    # Fix: slide.addText('「超越企业大学的方法论"", { -> slide.addText("「超越企业大学的方法论」", {
    content = content.replace("slide.addText('「超越企业大学的方法论\"\", {", 'slide.addText("「超越企业大学的方法论」", {')

    # Fix: slide.addText('「今天的结束，是行动的开始"", { -> slide.addText("「今天的结束，是行动的开始」", {
    content = content.replace("slide.addText('「今天的结束，是行动的开始\"\", {", 'slide.addText("「今天的结束，是行动的开始」", {')

    if content != original:
        with open(path, 'w', encoding='utf-8') as file:
            file.write(content)
        print(f'Fixed: {f}')

File: fix-mixed-quotes.js
const fs = require('fs');
const slidesDir = 'D:/新课开发/工作手册/专家隐性技能显性化/完整课程包/03-PPT演示课件/slides';
const files = fs.readdirSync(slidesDir).filter(f => f.startsWith('slide-') && f.endsWith('.js'));

files.forEach(file => {
  const filePath = slidesDir + '/' + file;
  let content = fs.readFileSync(filePath, 'utf8');

  // Fix pattern: \"Chinese text\「 -> 「Chinese text」
  // This is broken - appears to be Chinese text with mismatched escape sequences
  content = content.replace(/\\"([一-鿿]+[^\\]*)\\"\「/g, '「$1」');

  // Fix pattern: addText("...\ <- this is incomplete
  // Pattern: "text\「 where \ is a broken escape
  content = content.replace(/([一-鿿]+)\\"\「/g, '$1」');

  // Fix any remaining \「 that should just be 」 (when preceded by Chinese text)
  content = content.replace(/([一-鿿]+)\\"\「/g, '$1」');

  // Fix pattern: \"text without proper closing - just remove the escapes
  content = content.replace(/\\"([一-鿿]+)/g, '「$1');

  // Fix remaining broken escaped quotes
  content = content.replace(/\\"([^\\]*)"/g, (match, inner) => {
    if (/[一-鿿]/.test(inner)) {
      return '「' + inner + '」';
    }
    return '"' + inner + '"';
  });

  fs.writeFileSync(filePath, content, 'utf8');
});

console.log('Fixed', files.length, 'files');
